### PR TITLE
Run sim test for merge at epoch 0

### PR DESCRIPTION
**Pull request by dapplion**
**Motivation**

Part of #3275

**Description**

- Add ExecutionEngineMock
- getBlockRoot fast path for head block
- Initialize genesis block execution payload on mock
- Run sim test for merge at epoch 0

**TODO**
- Test currently breaks with a UnionType issue

```
TypeError: Cannot read property 'struct_hashTreeRoot' of undefined
    at UnionType.struct_getRootAtChunkIndex (/home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/union.ts:252:33)
    at UnionType.struct_yieldChunkRoots (/home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/abstract.ts:145:18)
    at struct_yieldChunkRoots.next (<anonymous>)
    at Function.from (<anonymous>)
```

### CODE REVIEW

**Comment by dapplion** [comment]
Persistent error on first block at slot 1

```
Eph 0/1 0.062 [NODE-A NODE-A API] error: Req req-z produceBlockV2 error message=Inconsistent execution payload parentHash
Error: Inconsistent execution payload parentHash
    at processExecutionPayload (/home/lion/Code/eth2.0/lodestar/packages/beacon-state-transition/src/merge/block/processExecutionPayload.ts:43:13)
    at Object.processBlock [as merge] (/home/lion/Code/eth2.0/lodestar/packages/beacon-state-transition/src/merge/block/index.ts:21:28)
    at Object.processBlock (/home/lion/Code/eth2.0/lodestar/packages/beacon-state-transition/src/allForks/stateTransition.ts:101:29)
    at computeNewStateRoot (/home/lion/Code/eth2.0/lodestar/packages/lodestar/src/chain/factory/block/index.ts:70:12)
    at assembleBlock (/home/lion/Code/eth2.0/lodestar/packages/lodestar/src/chain/factory/block/index.ts:53:21)
    at Object.produceBlock (/home/lion/Code/eth2.0/lodestar/packages/lodestar/src/api/impl/validator/index.ts:144:21)
    at Object.handler (/home/lion/Code/eth2.0/lodestar/packages/api/src/server/utils/server.ts:70:23)
```

Other errors

```
Eph 0/3 0.058 [NODE-A NODE-A API] error: Req req-1v produceBlockV2 error message=Cannot read property '0' of undefined
TypeError: Cannot read property '0' of undefined
    at ByteVectorType.struct_getRootAtChunkIndex (/home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/array.ts:104:53)
    at ByteVectorType.struct_yieldChunkRoots (/home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/abstract.ts:145:18)
    at struct_yieldChunkRoots.next (<anonymous>)
    at ByteVectorType.struct_convertToTree (/home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/array.ts:127:16)
    at /home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/container.ts:274:30
    at Array.map (<anonymous>)
    at ContainerType.struct_convertToTree (/home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/container.ts:270:37)
    at Object.set (/home/lion/Code/eth2.0/lodestar/packages/beacon-state-transition/src/allForks/util/cachedBeaconState.ts:436:22)
    at processExecutionPayload (/home/lion/Code/eth2.0/lodestar/packages/beacon-state-transition/src/merge/block/processExecutionPayload.ts:78:37)
    at Object.processBlock [as merge] (/home/lion/Code/eth2.0/lodestar/packages/beacon-state-transition/src/merge/block/index.ts:21:28)
    at Object.processBlock (/home/lion/Code/eth2.0/lodestar/packages/beacon-state-transition/src/allForks/stateTransition.ts:101:29)
    at computeNewStateRoot (/home/lion/Code/eth2.0/lodestar/packages/lodestar/src/chain/factory/block/index.ts:70:12)
    at assembleBlock (/home/lion/Code/eth2.0/lodestar/packages/lodestar/src/chain/factory/block/index.ts:53:21)
    at Object.produceBlock (/home/lion/Code/eth2.0/lodestar/packages/lodestar/src/api/impl/validator/index.ts:144:21)
    at Object.handler (/home/lion/Code/eth2.0/lodestar/packages/api/src/server/utils/server.ts:70:23)
```

**Comment by twoeths** [comment]
```
TypeError: Cannot read property 'struct_hashTreeRoot' of undefined
    at UnionType.struct_getRootAtChunkIndex (/home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/union.ts:252:33)
    at UnionType.struct_yieldChunkRoots (/home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/abstract.ts:145:18)
    at struct_yieldChunkRoots.next (<anonymous>)
    at Function.from (<anonymous>)
```
this is maybe fixed by #3285

**Comment by dapplion** [comment]
> ```
> TypeError: Cannot read property 'struct_hashTreeRoot' of undefined
>     at UnionType.struct_getRootAtChunkIndex (/home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/union.ts:252:33)
>     at UnionType.struct_yieldChunkRoots (/home/lion/Code/eth2.0/lodestar/node_modules/@chainsafe/ssz/src/types/composite/abstract.ts:145:18)
>     at struct_yieldChunkRoots.next (<anonymous>)
>     at Function.from (<anonymous>)
> ```
> 
> this is maybe fixed by #3285

@tuyennhv feel free to rebase, merged

File: packages/lodestar/src/chain/factory/block/body.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ export async function assembleBody(
       feeRecipient
     );
 
-    if (payloadId) {
+    if (payloadId !== null) {
```
Comment (dapplion) [inline packages/lodestar/src/chain/factory/block/body.ts:96]: Nice catch!!

**Pull request merged**